File: DSAlgorithms/Leetcode/c++/2_pointers/881-boats.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
       sort(people.begin(), people.end());
       int result = 0;
       int left = 0;
       int right = people.size() - 1;
       while(left <= right)
       {
           int remain = limit - people[right];
           right--;
           result+= 1;
           if(left <= right && remain >= people[left])
                left++;
       }
       return result;
    }
};

File: dsa_algo/leetcode/ts/src/2_pointers/881-boats.ts
const numRescueBoats = (people: number[], limit: number): number =>{
    people.sort((a, b) => a - b);
    let result = 0;
    let left = 0;
    let right = people.length - 1;
    while(left <= right)
    {
        let remain = limit - people[right];
        right --;
        result++;
        if(left <= right && remain >= people[left])
            left++;
    }
    return result;
};